### PR TITLE
feat(tests): add a CI ratchet for undocumented no-op input validation

### PR DESCRIPTION
**Pull request by somethingwithproof**
## Summary
`CactiValidator::validateInput()`/`form_input_validate()` accept an empty constraint/regex as a legitimate way to say a field is genuinely free text -- but that's indistinguishable in the code from validation having been skipped by mistake. An adjacent-code security review this cycle repeatedly found the second case in practice (graph-item `dashes`/`alpha`/`alpha2`/`gradheight`, automation `field`, `rrd_maximum`, and others all reached a shell/SQL sink because their `validateInput()` call had empty constraints).

Adds `tests/tools/validator_constraint_audit.php`, mirroring the existing `sql_interpolation_audit.php` ratchet:
- Tokenizer-based (not regex), so multi-line calls are read the way PHP reads them.
- `--check` wired into CI (new `validator-constraint-ratchet` job), `--write-baseline` to refresh.
- A `// no-validation: <reason>` comment on or immediately before a call exempts it, so genuinely-fine free-text fields don't need a manufactured constraint.
- Baseline grandfathers the current 250 undocumented sites (29 files) so this blocks *new* instances without demanding an immediate migration.

## Test plan
- New `tests/Unit/ValidatorConstraintAuditTest.php` (15 tests): classification of constrained/empty/documented-empty/dynamic for both functions, comment-adjacency edge cases, multi-call snippets.
- New `tests/integration/ValidatorConstraintRatchetIntegrationTest.php` (4 tests): CLI runs against the real tree, JSON output shape, baseline-matches-tree, baseline paths exist.
- `php -l` and `php-cs-fixer` clean.

### CODE REVIEW

File: tests/tools/validator_constraint_audit.php [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+			if ($comment_line >= $call_line - 1 && preg_match(NO_VALIDATION_COMMENT, $tok[1])) {
+				return true;
+			}
+
+			if ($comment_line < $call_line - 1) {
+				break;
+			}
+
+			continue;
+		}
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/tools/validator_constraint_audit.php:268]: Fixed.

File: tests/tools/validator_constraint_audit.php [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+				!is_array($tokens[$p - 1]) || $tokens[$p - 1][1] !== 'CactiValidator') {
+				continue;
+			}
+		}
+
+		$j = $i + 1;
+
+		while ($j < $n && is_array($tokens[$j]) && ($tokens[$j][0] === T_WHITESPACE || $tokens[$j][0] === T_COMMENT)) {
+			$j++;
+		}
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/tools/validator_constraint_audit.php]: Fixed.

File: tests/Unit/ValidatorConstraintAuditTest.php [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	))->toBe('documented-empty');
+});
+
+test('a no-validation comment on an unrelated earlier line does not exempt a later call', function () {
+	expect(validator_category_of(
+		"// no-validation: this refers to something else entirely\n" .
+		"\$other = 1;\n" .
+		"CactiValidator::validateInput(gnrv('notes'), 'notes', [], 3);"
+	))->toBe('empty');
+});
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/Unit/ValidatorConstraintAuditTest.php:71]: Fixed — added the trailing-comment-on-previous-line test case.

**Comment by somethingwithproof** [comment]
Fixed: `split_call_args()` now tracks `[]`/`{}` depth alongside parens so commas inside a constraints array no longer split the outer call's arguments.

**Comment by somethingwithproof** [comment]
Superseded by #7516, which merges this together with #7497. Closing in favor of the combined PR.
